lms/djangoapps/courseware/tests/test_submitting_problems.py

```diff
@@ -594,6 +598,46 @@ def test_dropping_all_correct(self):
         self.assertEqual(self.earned_hw_scores(), [1.0, 2.0, 2.0])  # Order matters
         self.assertEqual(self.score_for_hw('homework3'), [1.0, 1.0])
 
+    def test_min_grade_credit_requirements_status(self):
+        """
+        Test for credit course. If user passes minimum grade requirement then
+        status will be updated as satisfied in requirement status table.
+        """
+        self.basic_setup()
+        self.submit_question_answer('p1', {'2_1': 'Correct'})
+        self.submit_question_answer('p2', {'2_1': 'Correct'})
+
+        # Enable the course for credit
+        credit_course = CreditCourse.objects.create(
+            course_key=self.course.id,
+            enabled=True,
+        )
+
+        # Configure a credit provider for the course
+        credit_provider = CreditProvider.objects.create(
+            provider_id="ASU",
+            enable_integration=True,
+            provider_url="https://credit.example.com/request",
+        )
+        credit_course.providers.add(credit_provider)
+        credit_course.save()
+
+        # Add a single credit requirement (final grade)
+        CreditRequirement.objects.create(
+            course=credit_course,
+            namespace="grade",
+            name="grade",
+            criteria={"min_grade": 0.52}
+        )
```

Comment:
Please use the API call for this rather than directly manipulating Django models.